### PR TITLE
Add example dag and system test for S3ToGCSOperator

### PR DESCRIPTION
**Pull request by ephraimbuddy**
This PR closes one of the issues in #8280

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/master/UPDATING.md).

### CODE REVIEW

**Review by potiuk — APPROVED** [review]
:heart: Thanks Ephraim!

**Comment by potiuk** [comment]
Some static checks :)

**Comment by ephraimbuddy** [comment]
> Some static checks :)

Yeah. Addressing it now :)

File: docs/operators-and-hooks-ref.rst [hunk 1]
```diff
@@ -934,11 +934,15 @@ These integrations allow you to copy data from/to Google Cloud.
      - :doc:`How to use <howto/operator/amazon/aws/google_api_to_s3_transfer>`
      - :mod:`airflow.providers.amazon.aws.transfers.google_api_to_s3`
 
-   * - `Amazon Simple Storage Service (S3) <https://aws.amazon.com/s3/>`__
+   * - Any Data Storage service
```
Comment (ephraimbuddy) [inline docs/operators-and-hooks-ref.rst:937]: I'm not sure the best way to word this but I see that Google Transfer service is not only for the S3. https://cloud.google.com/storage-transfer-service.

Also, having it as S3 to GCS transfer will overshadow the S3ToGCSOperator. That's the reason I changed it and added the S3 to GCS below for S3ToGCSOperator

Comment (ephraimbuddy) [reply docs/operators-and-hooks-ref.rst]: I have removed this entirely. It's a duplicate. It's covered in google cloud operators as Storage Transfer Service and in Cloud Transfer Operators as s3 to gcs.

Comment (ephraimbuddy) [reply docs/operators-and-hooks-ref.rst]: https://github.com/apache/airflow/pull/10951/files#diff-471d3adab9e55d525a899f46345f3b2fL880-R884

**Comment by ephraimbuddy** [comment]
Hi @potiuk , I can't seem to figure out this pylint error. I will appreciate your assistance 🙂. Thanks

**Comment by ephraimbuddy** [comment]
All tests have passed🚀🚀🚀🚀

**Review by potiuk — APPROVED** [review]

**Comment by potiuk** [comment]
Cool :) 👍

**Pull request merged**